### PR TITLE
Logger sort of fixed

### PR DESCRIPTION
**Pull request by canyon289**
llmv_routes is not using the configured logger, detailed in the issue below. Still trying to figure it out

https://github.com/aivillage/llm_verification/issues/21

### CODE REVIEW

File: llmv_routes.py [hunk 1]
```diff
@@ -39,6 +41,12 @@ def llm_verification_index():
     @authed_only
     def generate_for_challenge():
         """Add a route to CTFd for generating text from a prompt."""
+        log.error("Log name %s, %s", log.name, log.level)
+        log.info("Received test generation request but his logger is not working for some reason")
+        log.error("NEW Received test generation request. Testing error level logging too")
```
Comment (canyon289) [inline llmv_routes.py:46]: For some reason info level logs don't print to console which is the main issue
image

Comment (canyon289) [reply llmv_routes.py:46]: Something funny is going on where the llmv_routes logging is going to the wrong logger

image

Comment (canyon289) [reply llmv_routes.py:46]: Maybe this. I'll figure it out tomorrow

https://stackoverflow.com/questions/27335686/adding-blueprints-to-flask-seems-to-break-logging

File: __init__.py [hunk 2]
```diff
@@ -11,13 +11,12 @@
 from .llmv_models import LlmSubmissionChallenge, fill_models_table
 from .llmv_routes import add_routes
 
-log = getLogger(__name__)
+log = initialize_llmvctfd_loggers(__name__)
```
Comment (canyon289) [inline __init__.py:14]: This fixes an issue where other modules were not logging to console

File: remote_llm.py [hunk 3]
```diff
@@ -32,15 +35,20 @@ def generate_text(preprompt, prompt, model):
...
     
-    log.info(f'Received text generation request for prompt "{prompt}" for model {model}')
+    log.error(f'Received text generation request for prompt "{prompt}" for model {model} at url {url}')
     # Load the Vanilla Neox API key from the config file.
 
     raw_response = post(url=route,
                         headers={'Authorization': f'Bearer {token}'},
                         json={'prompt': prompt, "preprompt" : preprompt, "model": model})
+
+    try:
```
Comment (canyon289) [inline remote_llm.py:45]: Provides a much better exception message than the json decode I was getting before with the bad llmv_router

**Comment by canyon289** [comment]
Ready for review. PR doesn't smoothly fix logging issue, just changes everything to error so it logs to console. It does add an extra exception handler which which provides readable exception of what is going wrong where

I propose merging and then fixing logging handler issues in a separate PR.

**Comment by comath** [comment]
The logger seemed to work fine on my machine 😁. I think the solution to this is a new function in `llm_logger` that returns the appropriate logger, guaranteed.

I'm not a fan of logging to error, but the rest looks good

**Comment by canyon289** [comment]
> The logger seemed to work fine on my machine 😁. I think the solution to this is a new function in `llm_logger` that returns the appropriate logger, guaranteed.
> 
> I'm not a fan of logging to error, but the rest looks good

Tried this but it seems to make things worse. Leaving this branch open for now, moved the json exception handling to another PR since thats the bigger problem anyway

https://github.com/aivillage/llm_verification/pull/22

=

**Pull request closed without merging**